File: templates/nim/aoctests.nim
import unittest
import aoc

suite "Advent of Code in Nim":

  test "parseInput works for sample data":
    check(parseInput("""
1x2x3
987x10x1
""") == @[
      Box(l: 1, w: 2, h: 3),
      Box(l: 987, w: 10, h: 1),
    ])

  test "part1 returns the sum of the volumes of the boxes":
    const boxes = @[Box(l: 1, w: 2, h: 3), Box(l: 1, w: 1, h: 1)]
    check part1(boxes) == 7

File: templates/csharp/AoCTests.cs
using Xunit;

public class AoCTests
{
    static readonly string[] Sample = ["1x2x3", "987x10x1"];

    [Fact]
    public void ParseInput_WorksForSample() =>
        Assert.Equal([new Box(1, 2, 3), new Box(987, 10, 1)],
                     AoC.ParseInput(Sample));

    [Fact]
    public void Part1_ReturnsTheSumOfTheVolumes() =>
        Assert.Equal(6 + 9870, AoC.Part1(AoC.ParseInput(Sample)));
}